src/v/coproc/tests/utils/wasm_event_generator.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+#include "bytes/bytes.h"
+#include "coproc/wasm_event.h"
+#include "model/record.h"
+#include "model/record_batch_reader.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <optional>
+
+/// \brief Pass this to a generator to recieve a genned event type of your
+/// choice
+enum class make_event_type { deploy, remove, random };
+
+/// Convienent struct for bundling together data necessary to serialized a
+/// wasm_event into a model::record. Any field set to std::nullopt will be
+/// skipped during serialization
+struct wasm_event {
+    using opt_action_t = std::optional<coproc::wasm_event_action>;
+    using opt_str_t = std::optional<ss::sstring>;
+    using opt_bytes_t = std::optional<bytes>;
+    opt_str_t uuid;
```

Comment:
what is `uuid` who generates this ?

Reply:
probably better named, 'name'

Reply:
but don't you have a name which is the sha256?

Reply:
name or previously called 'uuid', is the unique script id, and sha256 previously called checksum is the sha256 of the value